rest/server_context_test.go

```diff
@@ -325,3 +325,45 @@ outerLoop:
 
 	assert.True(t, existsOneMatchingEndpoint)
 }
+
+func TestObtainManagementEndpointsFromServerContextWithX509(t *testing.T) {
+	tb, teardownFn, caCertPath, certPath, keyPath := setupX509Tests(t, true)
+	defer tb.Close()
+	defer teardownFn()
+
+	original := tempConnectionDetailsForManagementEndpoints
+	defer func() {
+		tempConnectionDetailsForManagementEndpoints = original
+	}()
+
+	tempConnectionDetailsForManagementEndpoints = func() (string, string, string, string, string, string) {
+		return base.UnitTestUrl(), base.TestClusterUsername(), base.TestClusterPassword(), certPath, keyPath, caCertPath
+	}
+
+	ctx := NewServerContext(&ServerConfig{})
+	defer ctx.Close()
+
+	eps, err := ctx.ObtainManagementEndpoints()
+	assert.NoError(t, err)
+
+	baseSpec, err := connstr.Parse(base.UnitTestUrl())
+	require.NoError(t, err)
+
+	spec, err := connstr.Resolve(baseSpec)
+	require.NoError(t, err)
+
+	existsOneMatchingEndpoint := false
+
+outerLoop:
+	for _, httpHost := range spec.HttpHosts {
+		for _, ep := range eps {
+			formattedHttpHost := fmt.Sprintf("https://%s:%d", httpHost.Host, httpHost.Port)
```

Comment:
Is it possible to try and send a whoami request to this endpoint?
I think that would verify whether we need additional work around the tls.Config or not.

If this is already planned for a future PR that also makes sense.

Reply:
Yeah actually making a request is planned for the next PR.